lib/bolt/config.rb

```diff
@@ -34,50 +39,84 @@ class Config
                                     "targets on the command line and from plans.",
       "log"                      => "The configuration of the logfile output. Configuration can be set for "\
                                     "`console` and the path to a log file, such as `~/.puppetlabs/bolt/debug.log`.",
       "modulepath"               => "An array of directories that Bolt loads content (e.g. plans and tasks) from.",
-      "plugin_hooks"             => "Which plugins a specific hook should use.",
-      "plugins"                  => "A map of plugins and their configuration data.",
-      "puppetdb"                 => "A map containing options for configuring the Bolt PuppetDB client.",
-      "puppetfile"               => "A map containing options for the `bolt puppetfile install` command.",
-      "save-rerun"               => "Whether to update `.rerun.json` in the Bolt project directory. If "\
-                                    "your target names include passwords, set this value to `false` to avoid "\
-                                    "writing passwords to disk.",
-      "transport"                => "The default transport to use when the transport for a target is not specified "\
-                                    "in the URL or inventory.",
       "trusted-external-command" => "The path to an executable on the Bolt controller that can produce "\
                                     "external trusted facts. **External trusted facts are experimental in both "\
                                     "Puppet and Bolt and this API may change or be removed.**"
     }.freeze
 
+    # A combined map of all configuration options that can be set in this class.
+    # Includes all options except 'inventory-config', which is munged when loading
+    # a bolt-defaults.yaml file.
+    OPTIONS = BOLT_CONFIG.merge(INVENTORY_CONFIG).merge(PROJECT_CONFIG).freeze
+
+    # Default values for select options. These do not set the default values in Bolt
+    # and are only used for documentation.
     DEFAULT_OPTIONS = {
-      "color" => true,
+      "color"               => true,
       "compile-concurrency" => "Number of cores",
-      "concurrency" => "100 or one-third of the ulimit, whichever is lower",
-      "format" => "human",
-      "hiera-config" => "Boltdir/hiera.yaml",
-      "inventoryfile" => "Boltdir/inventory.yaml",
-      "modulepath" => ["Boltdir/modules", "Boltdir/site-modules", "Boltdir/site"],
-      "save-rerun" => true
+      "concurrency"         => "100 or one-seventh of the ulimit, whichever is lower",
+      "format"              => "human",
+      "hiera-config"        => "Boltdir/hiera.yaml",
+      "inventoryfile"       => "Boltdir/inventory.yaml",
+      "modulepath"          => ["Boltdir/modules", "Boltdir/site-modules", "Boltdir/site"],
+      "save-rerun"          => true,
+      "transport"           => "ssh"
+    }.freeze
+
+    PUPPETDB_OPTIONS = {
```

Comment:
I wonder if constants like this that are only defined for generating docs should be in a `Bolt::Config::Constants` class? Or `Bolt::Config::Options`?

Reply:
Maybe it would make sense to keep a single hash defining all the metadata about the options (documentation _and_ default) and then have the other constants be arrays of which keys are used in which context? @beechtom think of all the `.slice`ing!

Reply:
Like this?

```ruby
    # Options that configure Bolt. These options are used in bolt.yaml and bolt-defaults.yaml.
    BOLT_CONFIG = {
      "color" => {
        desc: "Whether to use colored output when printing messages to the console.",
        type: TrueClass,
        exmp: false,
        def: true
      },
      "compile-concurrency" => {
        desc: "The maximum number of simultaneous manifest block compiles.",
        type: Integer,
        exmp: 4,
        def: "Number of cores."
      },
```